### PR TITLE
[CAL-845] Improve the custom event name modal UI/UX

### LINKED ISSUE #6592: [CAL-845] Improve the custom event name modal UI/UX
Currently:

Should be

[Event type / Advanced \[Edit event name\] - Cal - Live (Figma)](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=20%3A24821&t=H83Rf75IpkiqPZuD-1)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-845](https://linear.app/calcom/issue/CAL-845/improve-the-custom-event-name-modal-uiux)</sub>

### PR DESCRIPTION
**Pull request by gitstart-calcom**
## What does this PR do?

Fixes #6592

## Demo
https://www.loom.com/share/3a4befcf3f0244e3a093e4fd4bdd6282

Screenshot 2023-01-24 at 3 03 29 PM

**Environment**: Staging(main branch) / Production

### CODE REVIEW

**Comment by PeerRich** [comment]
I think the idea was that the preview will update, no @Jaibles ?

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 1]
```diff
@@ -384,23 +384,41 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
...
+                  <div className="text-gray-900">
+                    <p className="mb-2">{t("event_name_info")}</p>
+                    <p className="mb-2">{t("your_full_name")}</p>
+                    <p className="mb-2">{t("scheduler_full_name")}</p>
+                    <p>{t("location_info")}</p>
+                  </div>
+                </div>
+              </div>
+              <h1 className="mb-1 font-medium">{t("preview")}</h1>
+              <img alt={t("preview")} src="/api/app-store/zapier/CalendarPreview.svg" />
```
Comment (PeerRich) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:413]: I think this was meant to update, not be an SVG. Also wrong folder

**Comment by Jaibles** [comment]
@PeerRich @gitstart-calcom yes, sorry for not being more explicit - the preview should take the real content. 

Obviously, in the case of the scheduler, that won't be available so it should just use what's in the table so should just read:
**Design chat between Alex Fisher and {Scheduler}**

**Review by emrysal — DISMISSED** [review]
Changes requested by @PeerRich and a conflict to resolve, marking accordingly.

**Comment by gitstart-calcom** [comment]
Hi @Jaibles, what should be the behavior of the Preview text? Should it update just after a user finishes editing the event name or after the event is saved (current behavior)? It would make more sense to have the preview update just after the text change. What do you think? cc: @PeerRich

**Comment by gitstart-calcom** [comment]
> > Hi @Jaibles, what should be the behavior of the Preview text? Should it update just after a user finishes editing the event name or after the event is saved (current behavior)? It would make more sense to have the preview update just after the text change. What do you think? cc: @PeerRich
> 
> Has to be after text change.

Alright, will have after text change implemented. Thanks

>  If it's after save they won't' see it unless they open the modal (unless I'm mistaken).

yes

**Comment by gitstart-calcom** [comment]
Hi @emrysal, PR has been updated please take a look. cc: @Jaibles @PeerRich

**Comment by PeerRich** [comment]
looks great! https://twitter.com/peer_rich/status/1618915636968828929

**Review by PeerRich — APPROVED** [review]

**Comment by PeerRich** [comment]
tested and works. waiting for @emrysal re-review of the code

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 2]
```diff
@@ -384,23 +410,63 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
...
+                    <p className="overflow-hidden text-ellipsis p-1.5 font-medium text-gray-900">
+                      {previewText}
+                    </p>
+                    <p className="ml-1.5 text-[10px] font-normal text-gray-600">8 - 10 AM</p>
+                  </div>
+                </div>
+              </div>
             </div>
             <DialogFooter>
+              <DialogClose onClick={() => formMethods.setValue("eventName", eventType.eventName ?? "")}>
```
Comment (emrysal) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:464]: This should happen in `onOpenChange` - and also unregister instead of defaulting

**Review by emrysal — CHANGES_REQUESTED** [review]

**Review by emrysal — APPROVED** [review]
Deemed my earlier comment a NIT

**Comment by PeerRich** [comment]
thank you for the great work!

**Pull request merged**